x/stakeibc/keeper/records.go

```diff
@@ -70,18 +70,15 @@ func (k Keeper) CleanupEpochUnbondingRecords(ctx sdk.Context, epochNumber uint64
 		for _, hostZoneUnbonding := range hostZoneUnbondings {
 			// if an EpochUnbondingRecord has any HostZoneUnbonding with non-zero balances, we don't delete the EpochUnbondingRecord
 			// because it has outstanding tokens that need to be claimed
-			if !hostZoneUnbonding.NativeTokenAmount.Equal(sdkmath.ZeroInt()) {
+			notClaimable := hostZoneUnbonding.Status != recordstypes.HostZoneUnbonding_CLAIMABLE
+			hasUnclaimedTokens := !hostZoneUnbonding.ClaimableNativeTokens.Equal(sdkmath.ZeroInt())
```

Comment:
Do you have to take into account the off by one error again here or is it always in the direction to short the user, never to leave an extra token behind so the absolute compare to zero is fine?

Reply:
To clarify the off by one error - it actually happens entirely on the host zone and it's materialized in a difference between our internal representation of the host zone and the ground truth.

To walk through an example:
* Let's say we intend to unbond 100uatom
* We have 100uatom stored on our EURs 
* We submit an undelegation for 100uatom
* The hub receives the message for 100uatom and processes it
* In our ack, we decrement whatever we need to by 100uatom
* 21 days later, only 99uatom land in our delegation ICA account after the unbonding finishes
* On stride, we still think we unbonded 100uatom <- here's where the discrepancy is

So this is resolved by two things:
1. We send a little extra atom to our delegation account when we register a host zone so that when we try and "sweep" 100uatom after the unbonding, it's able to grab the missing 1uatom from our initial funding
2. We submit slash/calibration queries that will update the total delegation on the host zone and validators to whatever the true value is (as determined by the query)

Reply:
If it's truly always off by 1utoken and unavoidable as part of the SDK, funding the delegation account with a little extra at the start seems like a good solution - covers us for 1 million tries!